geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/GroupBy.scala

```diff
@@ -0,0 +1,115 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+case class GroupBy[T](attribute: Int, exampleStat: String, sft: SimpleFeatureType)(implicit ct: ClassTag[T]) extends Stat {
+
+  override type S = GroupBy[T]
+
+  private [stats] val groupedStats: mutable.HashMap[T, Stat] = mutable.HashMap[T, Stat]()
+
+  def size: Int = groupedStats.size
+  def get(key: T): Option[Stat] = groupedStats.get(key)
+  def getOrElse[U >: Stat](key: T, default: => U = null): U = groupedStats.getOrElse(key, default)
```

Comment:
i think this shouldn't have a default value for `default` - doesn't really make sense, and could lead to bugs (see comment in `isEquivalent`). You could use `get().orNull` if that's what you want.